### PR TITLE
raw data sending

### PR DESCRIPTION
**Pull request by YasserRabee**

### CODE REVIEW

File: metrics/cadvisor.go [hunk 1]
```diff
@@ -107,40 +149,140 @@ func DecodeCAdvisor(r io.Reader) (CadvisorMetrics, error) {
...
+
+		})
 	}
-	podregexp := regexp.MustCompile(`pod[0-9a-f\-]+`)
-	podUID = podregexp.FindString(id)[3:]
-	if len(podUID) <= 0 {
-		return
+
+	nodes := cAdvisor.scanner.GetNodes()
+	pr, err := alltogether.NewConcurrentProcessor(nodes, getNodeMetrics)
```
Comment (mie00) [inline metrics/cadvisor.go]: @YasserRabee there is no need to use alltogether, let's use https://godoc.org/golang.org/x/sync/errgroup instead

File: utils/ticker.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		waitChan <- struct{}{}
+		close(waitChan)
+	}
+	ticker.waitChannels = make([]chan struct{}, 0)
+}
+
+// Start start scanner
+func (ticker *Ticker) Start(immediate, block bool) {
+	ticker.mutex.Lock()
+	defer ticker.mutex.Unlock()
```
Comment (mie00) [inline utils/ticker.go]: @YasserRabee this can result in deadlocks, please use use locking only when needed.

File: utils/utils.go [hunk 3]
```diff
@@ -194,3 +194,24 @@ func InSkipNamespace(skipNamespacePatterns []string, namespace string) bool {
...
+
+	return func(args ...interface{}) {
+		now := time.Now()
+		if now.After(nextTick) || now.Equal(nextTick) {
+			fn(args...)
+			nextTick = now.
+				Truncate(time.Second).
+				Truncate(interval).
+				Add(interval)
+			fmt.Println(nextTick.Format(time.RFC3339))
```
Comment (mie00) [inline utils/utils.go]: @YasserRabee if you still need this please use proper logging instead

Comment (YasserRabee) [reply utils/utils.go]: this is removed in later commit

File: utils/ticker.go (hunk 2 above)
Comment (YasserRabee) [reply utils/ticker.go]: this is fixed in later commit as well

File: metrics/cadvisor.go (hunk 1 above)
Comment (YasserRabee) [reply metrics/cadvisor.go]: @mie00 thanks. I will replace alltogether with errgroup. :+1:

**Pull request merged**